python/hail/matrixtable.py

```diff
@@ -526,34 +727,48 @@ def key_cols_by(self, *keys):
                 str_keys.append(k)
         return MatrixTable(self._jvds.keyColsBy(str_keys))
 
-    @typecheck_method(keys=oneof(str, Expression))
-    def key_rows_by(self, *keys):
+    @typecheck_method(keys=oneof(str, Expression),
+                      partition_key=nullable(oneof(oneof(str, Expression), listof(oneof(str, Expression)))))
+    def key_rows_by(self, *keys, partition_key=None):
         """Key rows by a new set of fields.
 
         Parameters
         ----------
         keys : varargs of :obj:`str`
             Row fields to key by.
+        partition_key : :obj:`str` or :obj:`list` of :obj:`str`, optional
```

Comment:
docs are wrong, takes expressions too. A bit annoying to write that out. Can you also fix the docs for keys above?